src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -250,5 +250,36 @@ public void ConnectionTestValidCredentialCombination()
 
             Assert.Equal(sqlCredential, conn.Credential);
         }
+
+        [Fact]
+        public void ConnectionTestWithCultureTH()
+        {
+            CultureInfo savedCulture = Thread.CurrentThread.CurrentCulture;
+            CultureInfo savedUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            try
+            {
+                Thread.CurrentThread.CurrentCulture = new CultureInfo("th-TH");
+                Thread.CurrentThread.CurrentUICulture = new CultureInfo("th-TH");
+
+                using TestTdsServer server = TestTdsServer.StartTestServer();
+                using SqlConnection connection = new SqlConnection(server.ConnectionString);
+                connection.Open();
+                Assert.Equal(ConnectionState.Open, connection.State);
+            }
+            catch
+            {
+                throw;
+            }
```

Comment:
This can be removed no?

Reply:
doesn't try/finally swallow the exception? At least it is what we thought. I did not test that. Let me do a quick test and will update here.

Reply:
Try..catch without rethrowing would swallow the exception. try..finally with no catch or filter clauses would still propogate the exception up the stack but you're guarenteed to have the finally clause run before the stack is unwound. This case, try..catch..throw is pointless.